### PR TITLE
Create ECR client and helper methods

### PR DESCRIPTION
**Pull request by MancunianSam**
The ECR client is now returning a cats.effect IO wrapper around the response and is using the AWS async client. The lambda consuming this is using cats-effect so this is fine but we may change this in the future.

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/aws/utils/ECRUtils.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package uk.gov.nationalarchives.aws.utils
+
+import software.amazon.awssdk.services.ecr.{EcrAsyncClient, EcrClient}
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/aws/utils/ECRUtils.scala]: Unnecessary import: `EcrClient`

File: src/main/scala/uk/gov/nationalarchives/aws/utils/ECRUtils.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import software.amazon.awssdk.services.ecr.model.{BatchGetImageRequest, DescribeImagesRequest, DescribeImagesResponse, DescribeRepositoriesResponse, ImageIdentifier, StartImageScanRequest, StartImageScanResponse}
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/aws/utils/ECRUtils.scala]: Unnecessary import: `BatchGetImageRequest`

File: src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala [hunk 3]
```diff
@@ -4,7 +4,9 @@ import java.net.URI
 
 import com.typesafe.config.{Config, ConfigFactory}
 import software.amazon.awssdk.http.apache.ApacheHttpClient
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient
 import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.ecr.{EcrAsyncClient, EcrClient}
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala]: Unnecessary import: `EcrClient`

File: src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala [hunk 4]
```diff
@@ -40,5 +42,14 @@ object Clients {
...
 
+  def ecr: EcrAsyncClient = {
+    val httpClient = NettyNioAsyncHttpClient.builder.build
+    EcrAsyncClient.builder
+      .region(Region.EU_WEST_2)
+      .endpointOverride(URI.create(configFactory.getString("ecr.endpoint")))
+      .httpClient(httpClient)
+      .build
+  }
+
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala]: Unnecessary line breaks

**Review by TomJKing — COMMENTED** [review]
Some minor comments

File: src/main/scala/uk/gov/nationalarchives/aws/utils/ECRUtils.scala (hunk 1 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/aws/utils/ECRUtils.scala]: I've cleared this one and the one below. I used the optimise imports on it so it's tidied up a lot of the awsskd imports.

File: src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala (hunk 3 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala]: I've taken this one out as well

File: src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala (hunk 4 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/aws/utils/Clients.scala]: I was doing so well with the line breaks up to this point :smile:

**Review by TomJKing — APPROVED** [review]

**Pull request merged**